exercises/pig-latin/src/example.c

```diff
@@ -0,0 +1,123 @@
+#include "pig_latin.h"
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <ctype.h>
+#include <stdbool.h>
+
+static bool is_vowel(const char letter)
+{
+   return letter == 'a' || letter == 'e' || letter == 'i' || letter == 'o'
+       || letter == 'u';
+}
+
+static bool starts_with_vowel(const char letter_a, const char letter_b)
+{
+   return is_vowel(letter_a) || (letter_a == 'x' && letter_b == 'r')
+       || (letter_a == 'y' && letter_b == 't');
+}
+
+static void count_words_and_chars(const char *input, size_t * words,
+                                  size_t * chars)
+{
+   bool in_word = false;
+   while (*input) {
+      (*chars)++;
+      if (isalpha(*input++)) {
+         *words += in_word ? 0 : 1;
+         in_word = true;
+      } else {
+         in_word = false;
+      }
+   }
+}
+
+static void mstrncpy(char *dst, const char *src, size_t n)
+{
+   while ((n--) && (*dst++ = tolower(*src++))) ;
+}
+
+static void append_ay(char *res)
+{
+   *res = 'a';
+   *(res + 1) = 'y';
+}
+
+static void process_pig_logic(char *dst, const char *src, size_t word_len)
+{
+   append_ay(dst + word_len);
+
+   const char letter_a = tolower(*src);
+   const char letter_b = letter_a ? tolower(*(src + 1)) : 0;
+
+   if (starts_with_vowel(letter_a, letter_b)) {
+      mstrncpy(dst, src, word_len);
+      return;
+   }
+
+   size_t i = 0;
+   while (src[i]) {
+      const char letter_a = tolower(src[i]);
+      const char letter_b = tolower(src[i + 1]);
+
+      if (is_vowel(letter_a))
+         break;
+
+      if (letter_b == 'y') {
+         i++;
+         break;
+      }
+
+      if (letter_a == 'q' && letter_b == 'u') {
```

Comment:
This is a dense solution so any trick you can use to make this easier to read is appreciated. In this case, how about using a helper like `starts_with(&src[i], "qu")`? I think that this could be helpful in a few places.